### PR TITLE
Cmake improve

### PR DESCRIPTION
**Pull request by Blackmane**
Improve cmake struct

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,9 +12,12 @@ The main purpose of this client is to be a desktop application fast and simple t
 - Qt min version 6.0.0
 - CMake min version 3.16.3
 
+```bash
+cmake -S . -B build && cmake --build build
```
Comment (guerinoni) [inline README.md:16]: This is not a substitute for other instruction builds...

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,15 +1,21 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.16.3)
 
-project(unofficial-discord-client LANGUAGES CXX)
+project(unofficial-discord-client
+	VERSION 1.0
+    DESCRIPTION "Lib to build its daemon"
```
Comment (guerinoni) [inline CMakeLists.txt:5]: VERSION and DESCRIPTION are not consistent

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
+    LANGUAGES CXX)
 
-set(CMAKE_INCLUDE_CURRENT_DIR ON)
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
```
Comment (guerinoni) [inline CMakeLists.txt:8]: Why this if? I don't understand...

File: CMakeLists.txt (hunk 2 above)
Comment (Blackmane) [reply CMakeLists.txt:5]: Good catch! My bad.

File: CMakeLists.txt (hunk 3 above)
Comment (guerinoni) [reply CMakeLists.txt:8]: This is an application, I don't think we need this now...

File: README.md (hunk 1 above)
Comment (Blackmane) [reply README.md:16]: Are you sure? Should be the same but just in one command line.

Comment (guerinoni) [reply README.md:16]: I prefer to keep simple as possible for new developer :)

Comment (Blackmane) [reply README.md:16]: So better split the two command?
Like:
```
cmake -S . -B build
cmake --build build 
```

Comment (guerinoni) [reply README.md:16]: Yes, better

**Review by guerinoni — CHANGES_REQUESTED** [review]

**Review by guerinoni — APPROVED** [review]
LGTM

**Pull request merged**